### ISSUE TITLE
Mutable indexing doesn't seem to work

### ISSUE THREAD

**Issue by nazar-pc** [ferrilab/bitvec#278]
I use `BitArray<[u8; _]>` as a container and then trying to index into it.

Immutable indexing to return `&bool` works just fine, but attempt to return `&mut bool` fails:
```
218 |         &mut self.encoded_record_chunks_used[index]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
    |
    = help: trait `IndexMut` is required to modify indexed content, but it is not implemented for `bitvec::array::BitArray<[u8; 8192]>`
```

Not sure what the reason of this is, but it is unexpected for sure.

**Comment by edgarogh** [comment]
It's impossible for a BitSlice to return a `&mut bool` through indexing since it doesn't contain any `bool` value. It only works with immutable indexing because it returns references to global `&false` and `&true` constant values.

**Comment by nazar-pc** [comment]
True, but it could return something that is `impl DerefMut<Output = bool>`, achieving mostly the same result

**Comment by edgarogh** [comment]
That might be a bit odd to use, but indeed, that could work.

**Comment by teor2345** [comment]
> True, but it could return something that is `impl DerefMut<Output = bool>`, achieving mostly the same result

I think it would also have to `impl Drop`. `DerefMut` would return a temporary `&mut bool`, but we want to write the mutated value back to the original bit location. And there's no way to write directly to a bit using a `&mut bool`.

So `bitvec` could run the bit writing code using `Drop`, or a "persist" method you have to call on the returned type.

**Comment by Timmmm** [comment]
I think this should be documented at least. I.e. how do you set single bits if you can't do this? Setting a single bit is surely a very common use case? (I think it's [`set`](https://docs.rs/bitvec/latest/bitvec/slice/struct.BitSlice.html#method.set)).
